thoth-core/src/main/java/fr/maif/projections/EventuallyConsistentProjection.java

```diff
@@ -0,0 +1,277 @@
+
+    @Builder(toBuilder = true)
+    @AllArgsConstructor(access = AccessLevel.PRIVATE)
+    public static class Config<E extends Event, Meta, Context> {
+        public final String topic;
+        public final String groupId;
+        public final String bootstrapServers;
+        public final Function<ConsumerSettings<String, EventEnvelope<E, Meta, Context>>, ConsumerSettings<String, EventEnvelope<E, Meta, Context>>> completeConfig;
+        public final Duration minBackoff;
+        public final Duration maxBackoff;
+        public final Integer randomFactor;
+        public final Integer commitSize;
+
+        public static <E extends Event, Meta, Context> Config<E, Meta, Context> create(String topic, String groupId, String bootstrapServers) {
+            return Config.<E, Meta, Context>builder()
+                    .bootstrapServers(bootstrapServers)
+                    .groupId(groupId)
+                    .topic(topic)
+                    .build();
+        }
+    }
+
+    protected final ActorSystem actorSystem;
+    protected final Materializer materializer;
+
+    protected final String topic;
+    protected final String groupId;
+    protected final String bootstrapServers;
+    protected final Duration minBackoff;
+    protected final Duration maxBackoff;
+    protected final double randomFactor;
+    protected final Integer commitSize;
+    protected final ConsumerSettings<String, EventEnvelope<E, Meta, Context>> consumerSettings;
+
+    protected final AtomicReference<Consumer.Control> controlRef = new AtomicReference<>();
+    protected final AtomicReference<Status> innerStatus = new AtomicReference<>(Status.stopped);
+
+    public EventuallyConsistentProjection(ActorSystem actorSystem, Config<E, Meta, Context> config) {
+        this.actorSystem = actorSystem;
+        this.materializer = Materializer.createMaterializer(actorSystem);
+        this.topic = config.topic;
+        this.groupId = config.groupId;
+        this.bootstrapServers = config.bootstrapServers;
+        this.minBackoff = Objects.isNull(config.minBackoff) ? Duration.ofSeconds(30) : config.minBackoff;
+        this.maxBackoff = Objects.isNull(config.maxBackoff) ? Duration.ofMinutes(30) : config.maxBackoff;
+        this.randomFactor = Objects.isNull(config.randomFactor) ? 0.2d : config.randomFactor;
+        this.commitSize = Objects.isNull(config.commitSize) ? 10 : config.commitSize;
+        Function<ConsumerSettings<String, EventEnvelope<E, Meta, Context>>, ConsumerSettings<String, EventEnvelope<E, Meta, Context>>> completeConfig = Objects.isNull(config.completeConfig) ? e -> e : config.completeConfig;
+        this.consumerSettings = completeConfig.apply(ConsumerSettings
+                .create(actorSystem, new StringDeserializer(), deserializer())
+                .withGroupId(groupId)
+                .withProperty(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "latest")
```

Comment:
Should we make this a config choice ? In some cases earliest may be best (to retrieve event of the past).

Reply:
it's possible via the function `ConsumerSetting -> ConsumerSetting`